### PR TITLE
Add KUBERNETES_SERVICE_DNS_DOMAIN to deployment templating

### PR DESCRIPTION
**Pull request by maltesander**
prerequisite for https://github.com/stackabletech/issues/issues/436#issuecomment-2413082601

### CODE REVIEW

File: template/deploy/helm/[[operator]]/templates/deployment.yaml.j2 [hunk 1]
```diff
@@ -48,6 +48,8 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.annotations['internal.stackable.tech/image']
+            - name: CLUSTER_DNS_DOMAIN
```
Comment (sbernauer) [inline template/deploy/helm/[[operator]]/templates/deployment.yaml.j2:51]: Any comments on how you got this name? I personally find `KUBERNETES_SERVICE_DNS_DOMAIN` more speaking (as it's the *service* domain, not the cluster domain I guess)

Comment (maltesander) [reply template/deploy/helm/[[operator]]/templates/deployment.yaml.j2:51]: No strong opinion, will change. Just did not want to use Kubernetes (due to openshift, although they have the env vars alla KUBERNETES_SERVICE_HOST as well... will change

Comment (maltesander) [reply template/deploy/helm/[[operator]]/templates/deployment.yaml.j2]: https://github.com/stackabletech/operator-templating/pull/445/commits/ea6761bbbd88b1729fb574bdf40dbf47b2c83348

Comment (sbernauer) [reply template/deploy/helm/[[operator]]/templates/deployment.yaml.j2]: Many thanks!

**Review by sbernauer — APPROVED** [review]

**Pull request merged**